File: digital_wallet_fastapi/app/models/transaction.py
from datetime import datetime
from decimal import Decimal

from sqlalchemy import BigInteger, Numeric, String, DateTime, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Transaction(Base):
    __tablename__ = "transactions"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    from_wallet_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    to_wallet_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    amount: Mapped[Decimal] = mapped_column(Numeric(18, 4), nullable=False)
    tx_type: Mapped[str] = mapped_column(String(20), nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, default="SUCCESS")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=False), server_default=func.now())

File: digital_wallet_laravel/app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    protected $table = 'transactions';

    protected $fillable = ['from_wallet_id', 'to_wallet_id', 'amount', 'tx_type', 'status'];

    public $timestamps = false;

    protected $casts = [
        'id' => 'integer',
        'from_wallet_id' => 'integer',
        'to_wallet_id' => 'integer',
        'amount' => 'decimal:4',
        'created_at' => 'datetime',
    ];
}
